### PR TITLE
Add `/metrics` endpoint

### PR DESCRIPTION
**Pull request by kmdkuk**
- Add `/metrics` endpoint as Prometheus metrics. 
  - Exported metrics are pod_status.
- Added an environment variable named RUNNER_POOL_NAME.
- Add Document for managed state and metrics

### CODE REVIEW

File: agent/metrics.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package agent
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "runner_pool"
```
Comment (masa213f) [inline agent/metrics.go:8]:
```suggestion
	namespace = "runner_"
```

File: agent/metrics.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package agent
```
Comment (masa213f) [inline agent/metrics.go:1]: Could you create `metrics` package?

File: agent/metrics.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+)
+
+func MetricsInit(registry prometheus.Registerer, name string) {
+	labels := prometheus.Labels{
+		"runner_pool_name": name,
+	}
+	podStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace:   namespace,
+			Name:        "pod",
```
Comment (masa213f) [inline agent/metrics.go:62]:
```suggestion
			Name:        "pod_status",
```

File: agent/metrics.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+			Help:        "`status` label show the status of the pod.",
+			ConstLabels: labels,
+		},
+		[]string{"status"},
+	)
+	jobStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace:   namespace,
+			Name:        "job",
```
Comment (masa213f) [inline agent/metrics.go:71]:
```suggestion
			Name:        "job_status",
```

File: agent/metrics.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	)
+}
+
+func UpdatePodStatus(label PodStatus) {
+	for _, labelStatus := range AllPodStatus {
+		if labelStatus == label {
+			podStatus.WithLabelValues(string(labelStatus)).Set(1)
+		} else {
+			podStatus.WithLabelValues(string(labelStatus)).Set(0)
+		}
```
Comment (masa213f) [inline agent/metrics.go:99]:
```suggestion
		val := 0
		if labelStatus == label {
			val = 1
		}
		podStatus.WithLabelValues(string(labelStatus)).Set(val)
```

File: config/samples/manifests.yaml [hunk 6]
```diff
@@ -17,6 +17,10 @@ spec:
       containers:
       - name: runner
         image: actions-runner:latest
+        ports:
+          - name: metrics
+            containerPort: 8080
+            protocol: TCP
```
Comment (zoetrope) [inline config/samples/manifests.yaml:23]: Do we always need to specify the metrics port in a custom resource?
Is it possible for the controller to automatically specify the port?

**Review by binoue — APPROVED** [review]
LGTM

File: config/samples/manifests.yaml (hunk 6 above)
Comment (masa213f) [reply config/samples/manifests.yaml:23]: That's good.

I plan to rewrite the reconcile loop significantly, in https://github.com/cybozu-go/github-actions-controller/pull/25.
I will adopt your suggestion there.

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**